src/tasks/withdraw.ts

```diff
@@ -83,15 +81,133 @@ async function fastTokenDetails(
   return erc20Token(address, hre);
 }
 
-async function getWithdrawals(
-  tokens: string[],
-  settlement: Contract,
-  minValue: string,
-  leftover: string,
-  hre: HardhatRuntimeEnvironment,
+interface ComputeSettlementInput {
+  withdrawals: Omit<Withdrawal, "gas">[];
+  receiver: string;
+  solverForSimulation: string;
+  settlement: Contract;
+  hre: HardhatRuntimeEnvironment;
+}
+async function computeSettlement({
+  withdrawals,
+  receiver,
+  solverForSimulation,
+  settlement,
+  hre,
+}: ComputeSettlementInput) {
+  const encoder = new SettlementEncoder({});
+  withdrawals.forEach(({ token, amount }) =>
```

Comment:
Just curious, is there a reason you prefer `array.forEach((x) => {...})` over `for (const x of array) {...}`?

Reply:
No reason! Maybe I did it with `map` first, realized I didn't need the output and just changed it to `forEach`.